src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/AvailableType.cs

```diff
@@ -11,11 +11,12 @@
 
 namespace ILCompiler.DependencyAnalysis.ReadyToRun
 {
-    public class AvailableType : DependencyNodeCore<NodeFactory>, IEETypeNode
+    public class AvailableType : PrecodeHelperImport, IEETypeNode
```

Comment:
What is relocing to `AvailableType` and `ExternalType` that requires this default precode import? I would imagine we'd be explicit when asking for one of these using the `ReadyToRunCodegenNodeFactory.CreateTypeHandleHelper` helper. I made these two derive from `DependencyNodeCore` because we don't emit types themselves currently with R2R.

Reply:
EH info used to. Apparently it no longer does after my change. I'll revisit this while consolidating the changes for PR.

Reply:
I have verified that making AvailableType and ExternalTypeNode relocable is no longer needed after my EH info fixes so I have reverted them locally and I'll push them to the PR with the next batch of changes.